js/src/components/shipping-rate-section/minimum-order-card/add-minimum-order-form-modal.js

```diff
@@ -14,8 +14,29 @@ import AppInputPriceControl from '.~/components/app-input-price-control/index.js
 import VerticalGapLayout from '.~/components/vertical-gap-layout';
 import AppCountrySelect from '.~/components/app-country-select';
 
-const AddMinimumOrderModal = ( props ) => {
-	const { countryOptions, value, onChange = noop, onRequestClose } = props;
+/**
+ * Display the add minimum order modal that is wrapped in a Form.
+ *
+ * When users submit the form, `props.onRequestClose` will be called first, and then followed by `props.onSubmit`.
+ * If we were to call `props.onSubmit` first, it may cause some state change in the parent component and causing this component not to be rendered,
+ * and when `props.onRequestClose` is called later, there would be a runtime React error because the component is no longer there.
+ *
+ * @param {Object} props Props.
+ * @param {Array<string>} props.countryOptions Array of country codes options, to be used as options in AppCountrySelect.
+ * @param {Object} props.initialValues Initial values for the form.
+ * @param {Array<string>} props.initialValues.countries Array of selected country codes.
+ * @param {string} props.initialValues.currency Selected currency.
+ * @param {number} props.initialValues.threshold Threshold value.
+ * @param {function()} props.onRequestClose Callback to close the modal.
+ * @param {function(Object)} props.onSubmit Callback when the form is submitted, with the form value.
+ */
+const AddMinimumOrderFormModal = ( props ) => {
+	const {
+		countryOptions,
+		initialValues,
+		onSubmit = noop,
+		onRequestClose = noop,
+	} = props;
 
```

Comment:
:nail_care: Just a personal preference, not tp have to think: "Is `props` going to be used somewhere below as well?"
```suggestion
const AddMinimumOrderFormModal = ( {
	countryOptions,
	initialValues,
	onSubmit = noop,
	onRequestClose = noop,
} ) => {
```

However, then we'd need 
```
		'jsdoc/check-param-names': [
			'warn',
			{ disableExtraPropertyReporting: false },
		],
```
to avoid complaints about documented non-destructed `props.initialValues`. (Which I find rather strange, as it didn't complain about just `props`)

Reply:
@tomalec , am I doing this correctly? I'm still seeing ESLint JSDoc complaining: 🤔 

![image](https://user-images.githubusercontent.com/417342/162487555-21881964-eb25-40c9-9493-a72d26f802fe.png)

I end up using another way, by using a new JSDoc typedef `MinimumOrderGroup`, in [`11ca55c` (#1410)](https://github.com/woocommerce/google-listings-and-ads/pull/1410/commits/11ca55c460539722dc72b4abb502f12570d55910).